src/main/java/seedu/address/model/AddressBook.java

```diff
@@ -130,6 +130,20 @@ public void deleteEvent(Event e) {
         this.events.remove(e);
     }
 
+    /**
+     * Returns true if {@code event} exists in this {@code AddressBook}.
+     */
+    public boolean hasEvent(Event event) {
+        requireNonNull(event);
+        return events.contains(event);
+    }
+
+    public void setEvent(Event target, Event editedEvent) {
+        requireNonNull(editedEvent);
+
+        events.setEvent(target, editedEvent);
+    }
```

Comment:
Perhaps you might want to add JavaDocs for the `setEvent` method similar to `setPerson`'s JavaDocs